### DISCUSSION TITLE
Modules as a git sub-moduel repository

### DISCUSSION THREAD

**Discussion by DharitParmar** [android/nowinandroid#1224] [General]
Can each modules define here live as a git submodules?

**Comment by SimonMarquis** [comment]
Hi @DharitParmar, yes they could be git submodules, but there would be not point of doing that. Did you have something specific in mind?

**Reply by DharitParmar** [reply]
Thanks for a quick response! appriciate it! It's just that today I use submodules in my app and that unables me to point to a diffirent commit for a different feature and also reusing it by diifferent team(s). So I was wondering 1] is that possible here? 2] how feasible to resue these core modules for kotlin multiplatform and plug it into iOS?

**Reply by SimonMarquis** [reply]
> 1] is that possible here?

yes, you can do it if you want to

> 2] how feasible to resue these core modules for kotlin multiplatform and plug it into iOS?

This is probably not the best place for this kind of discussion as this project is not meant for multiplatform at the moment. Although [Kotlin Multiplatform](https://kotlinlang.org/docs/multiplatform-get-started.html) is what you are looking for.

**Comment by lihenggui** [comment]
> 2] how feasible to resue these core modules for kotlin multiplatform and plug it into iOS?

You need to replace some core dependencies to support Kotlin multiplatform. For example: migrate Android Room to SqlDelight to support multiplatform in the `:core:database` module. There is a lot of work to do.
